ambry-store/src/test/java/com.github.ambry.store/StatsManagerTest.java

```diff
@@ -0,0 +1,217 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import org.junit.After;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+
+public class StatsManagerTest {
+  private final StatsManager statsManager;
+  private final Map<PartitionId, Store> storeMap;
+  private final String outputFileString;
+  private final File tempDir;
+
+  public StatsManagerTest() throws IOException, StoreException {
+    tempDir = StoreTestUtils.createTempDirectory("nodeStatsDir-" + UtilsTest.getRandomString(10));
+    outputFileString = (new File(tempDir.getAbsolutePath(), "stats_output.json")).getAbsolutePath();
+    long[][] testData1 = new long[][]{{100, 200, 300}, {100}, {200, 300}};
+    long[][] testData2 = new long[][]{{10}, {50}};
+    long[][] testData3 = new long[][]{{1000}, {200}, {300}, {10, 20, 30, 40, 50}};
+    List<StatsSnapshot> statsSnapshots = new ArrayList<>();
+    statsSnapshots.add(generateStatsSnapshot(testData1));
+    statsSnapshots.add(generateStatsSnapshot(testData2));
+    statsSnapshots.add(generateStatsSnapshot(testData3));
+    storeMap = new HashMap<>();
+    MetricRegistry registry = new MetricRegistry();
+    StorageManagerMetrics metrics = new StorageManagerMetrics(registry);
+    for (int i = 0; i < 3; i++) {
+      PartitionId id = new Partition(i, PartitionState.READ_WRITE, 1024 * 1024 * 1024);
+      Store store =
+          new MockBlobStore(UtilsTest.getRandomString(10), metrics, new MockBlobStoreStats(statsSnapshots.get(i)));
+      storeMap.put(id, store);
+    }
+    storeMap.put(new Partition(1000, PartitionState.READ_WRITE, 1024 * 1024 * 1024), null);
+    StorageManager storageManager = new MockStorageManager(storeMap);
+    Properties properties = new Properties();
+    properties.put("stats.output.file.path", outputFileString);
+    StatsManagerConfig config = new StatsManagerConfig(new VerifiableProperties(properties));
+    statsManager = new StatsManager(storageManager, new ArrayList<>(storeMap.keySet()), new MetricRegistry(), config,
+        SystemTime.getInstance());
+  }
+
+  /**
+   * Deletes the temporary directory.
+   * @throws InterruptedException
+   * @throws IOException
+   */
+  @After
+  public void cleanup() throws IOException {
+    assertTrue(tempDir.getAbsolutePath() + " could not be deleted", StoreTestUtils.cleanDirectory(tempDir, true));
+  }
+
+  /**
+   * Test to verify that the {@link StatsManager} is collecting, aggregating and publishing correctly using predefined
+   * data sets and mocked {@link Store}s and {@link StorageManager}.
+   * @throws StoreException
+   * @throws IOException
+   * @throws InterruptedException
+   */
+  @Test
+  public void testStatsManagerCollectAggregateAndPublish() throws StoreException, IOException, InterruptedException {
+    long[][] aggregatedTestData = new long[][]{{1110, 200, 300}, {350}, {500, 300}, {10, 20, 30, 40, 50}};
```

Comment:
hardCoding is not a good practice. You should try to calculate on the go or atleast have instance variable which tracks this value (which is set in setupTest())